string.go

```diff
@@ -30,40 +28,45 @@ import (
 	"unicode/utf8"
 )
 
-// AESEncrypt encrypts text and given key with AES.
-func AESEncrypt(key, text []byte) ([]byte, error) {
+// AESEncrypt encrypts plaintext with the given key and nonce using AES in GCM mode.
+func AESEncrypt(key, nonce, plaintext []byte) ([]byte, error) {
```

Comment:
Not automatically generating and appending the nonce leaves every caller with the task of generating the nonce itself, thus knowing about the actual encryption algorithm this function uses.
AES/GCM requires a 96 bit nonce (`gcm.NonceSize()`), other ciphers might require a different nonce length.
Please rename this function to `AESGCMEncrypt`.